crates/ruff/src/rules/ruff/rules/pairwise_over_zipped.rs

```diff
@@ -0,0 +1,117 @@
+use std::num::ParseIntError;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::source_code::Stylist;
+use rustpython_parser::ast::{Expr, ExprKind};
+
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+use crate::Range;
+use ruff_python_ast::helpers::unparse_constant;
+
+#[violation]
+pub struct PairwiseOverZipped {}
+
+impl Violation for PairwiseOverZipped {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Prefer `itertools.pairwise()` to `zip()` if trying to fetch an iterable's successive overlapping pairs.")
+    }
+}
+
+#[derive(Debug)]
+struct SliceInfo {
+    arg_name: String,
+    slice_start: Result<i64, ParseIntError>,
+    slice_end: Result<i64, ParseIntError>,
+}
+
+impl SliceInfo {
+    pub fn new(
+        arg_name: String,
+        slice_start: Result<i64, ParseIntError>,
+        slice_end: Result<i64, ParseIntError>,
```

Comment:
It seems that we only care about whether parsing was successful or not, but not what the actual parsing error was. That means we could use `Option<i64>` here (you can use `result.ok()` to get an option) and safe a few bytes of memory.

Reply:
Yeah, that's the path I'm going down in the latest commit. Seems like it'll help with the default issues that @andersk raised, since we can just default to `None` instead of a number.